e2e/test/iothub/properties/PropertiesWithComponentsE2ETests.cs

```diff
@@ -0,0 +1,502 @@
+            var actualProp = completeTwin.Properties.Reported[ComponentName][propName];
+            Assert.AreEqual(JsonConvert.SerializeObject(actualProp), JsonConvert.SerializeObject(propValue));
+        }
+
+        public static async Task<Task> SetClientPropertyUpdateCallbackHandlerAsync<T>(DeviceClient deviceClient, string expectedComponentName, string expectedPropName, T expectedPropValue, MsTestLogger logger)
+        {
+            var propertyUpdateReceived = new TaskCompletionSource<bool>();
+            string userContext = "myContext";
+
+            await deviceClient
+                .SubscribeToWritablePropertiesEventAsync(
+                    (patch, context) =>
+                    {
+                        logger.Trace($"{nameof(SetClientPropertyUpdateCallbackHandlerAsync)}: WritableProperty: {patch}, {context}");
+
+                        try
+                        {
+                            if (patch.TryGetValue<T>(expectedComponentName, expectedPropName, out var propertyFromCollection))
+                            {
+                                //Assert.AreEqual(JsonConvert.SerializeObject(JsonConvert.DeserializeObject<T>(propertyFromCollection.Replace("\\\"", ""))), JsonConvert.SerializeObject(expectedPropValue));
+                                Assert.AreEqual(JsonConvert.SerializeObject(expectedPropValue), JsonConvert.SerializeObject(propertyFromCollection));
+                            }
+                            else
+                            {
+                                Assert.Fail("Property was not found in the collection.");
+                            }
+                            Assert.AreEqual(userContext, context, "Context");
+                        }
+                        catch (Exception e)
+                        {
+                            propertyUpdateReceived.SetException(e);
+                        }
+                        finally
+                        {
+                            propertyUpdateReceived.SetResult(true);
+                        }
+
+                        return Task.FromResult<bool>(true);
+                    },
+                    userContext)
+                .ConfigureAwait(false);
+
+            return propertyUpdateReceived.Task;
+        }
+
+        public static async Task RegistryManagerUpdateWritablePropertyAsync<T>(string deviceId, string componentName, string propName, T propValue)
+        {
+            using var registryManager = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+
+            var twinPatch = new Twin();
+            twinPatch.Properties.Desired[componentName] = new
+            {
+                __t = "c",
+            };
+            if (propValue is List<object>)
+            {
+                twinPatch.Properties.Desired[componentName][propName] = (Newtonsoft.Json.Linq.JToken)(JsonConvert.DeserializeObject(JsonConvert.SerializeObject(propValue)));
```

Comment:
why do we need to serialize and deserialize it a list value? isn't Twin able to accept the object as-is?

Reply:
don't need this